Automattic/jetpack
# Blog Subscription Widget: deprecate widget and transform to block

#### Changes proposed in this Pull Request:

With the new current widget block editor, the Blog Subscription widget is redundant. The current Subscription Form block already exists and offers the same functions. This PR removes it from the `core/legacy-widget` and adds the widget to the REST API to allow a block `transforms:` to be created.

*This will preserve the following settings that exist in the block: show_subscribers_total, subscribe_button, subscribe_placeholder*.

#### Jetpack product discussion
Part of the widget => block project #21069 

#### Does this pull request change what data or activity we track or use?
Nope

#### Testing instructions:
* Before switching to this PR, go to Customizer ⇢ Widgets and add Legacy Widget choosing Blog Subscription from the dropdown. *Be sure to save changes*
* Next switch to this PR and view your customizer.
* You should see that Blog Subscription is no longer an option in the Legacy Widget block.
* Now click the previously added Legacy Blog Subscription and use the Transform button to change to the Subscription Form block.
